modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -643,3 +676,59 @@ class TrainingCanceledError extends Error {
     this.name = 'CancelError'
   }
 }
+function extractAugmentations(
+  intent: Readonly<{
+    name: string
+    contexts: string[]
+    slot_definitions: Readonly<{ name: string; entity: string }>[]
+    utterances: string[]
+    vocab?: _.Dictionary<boolean>
+    slot_entities?: string[]
+  }>,
+  complex_entities: Readonly<{
+    name: string
+    pattern_entities: string[]
+    list_entities: string[]
+    examples: string[]
+  }>[],
+  list_entities: Readonly<{
+    name: string
+    synonyms: { [canonical: string]: string[] }
+    fuzzyTolerance: number
+    sensitive: boolean
+  }>[],
+  pattern_entities: Readonly<{
+    name: string
+    pattern: string
+    examples: string[]
+    matchCase: boolean
+    sensitive: boolean
+  }>[]
+) {
+  return intent.slot_definitions
+    .map(slot => {
+      const complexEntity = complex_entities.find(x => x.name === slot.entity)
+
+      const listEntities = list_entities.filter(
+        x => x.name === slot.entity || complexEntity?.list_entities.includes(slot.entity)
+      )
+
+      const patternEntities = pattern_entities.filter(
+        x => x.name === slot.entity || complexEntity?.list_entities.includes(slot.entity)
+      )
+
+      const listExamples = interleave(
+        ...listEntities.map(x => interleave(...Object.keys(x.synonyms).map(key => [key, ...x.synonyms[key]])))
+      )
+
+      const patternExamples = interleave(...patternEntities.map(x => x.examples))
+
+      const complexExamples = complexEntity?.examples ?? []
+
+      return {
+        variableName: slot.name,
```

Comment:
we don't have any reference to variable in the while nlu code base, can you rename the key for slotName, it'll be easier to understand. so return type would be a. `SlotExamples` you can reuse that in your augmenter